katapult.go

```diff
@@ -53,13 +55,23 @@ func WithBaseURL(u *url.URL) Opt {
 }
 
 func WithAPIKey(key string) Opt {
-	return func(c *Client) error {
+	return func(c *Client, _ *http.Client) error {
 		c.APIKey = key
 
 		return nil
 	}
 }
 
+// WithTracing wraps the http client Transport with the otelhttp helper
+// This captures outgoing request details.
+func WithTracing() Opt {
+	return func(c *Client, httpClient *http.Client) error {
+		httpClient.Transport = otelhttp.NewTransport(httpClient.Transport)
+
+		return nil
+	}
+}
```

Comment:
I think we should probably support passing in a list of `otelhttp.Option...` funcs and pass them on to `otelhttp.NewTransport()`.

Considering this option effectively does nothing when the `WithHTTPClient()` option is used as it replaces the `http.Client` used, while this option configures the default client instance only.

I wonder if maybe we should return an error if both options are provide to avoid unexpected behavior?

The alternative way to avoid tracing unexpectedly not working, would be to have the tracing modify the client after initial options setup, but that can potentially worse side-effects by modifying a custom passed in `http.Client` which might be used elsewhere, causing even worse unexpected behavior... lol